### ISSUE TITLE
Sudden increase in Redis clients

### ISSUE THREAD

**Issue by itzloop** [livekit/livekit#2238]
**Describe the bug**
After upgrading from v1.5.0 to v1.5.1, We are seeing massive increases in Redis clients connecting to our Redis instances. by upgrading only a single instance to v1.5.1, we are seeing massive increase to Redis client count (from ~2000 up to ~25K). 

**Server**
- Version: [1.5.1]
- Environment: [Bare metal]
We are running a 3 node Livekit cluster and a replicated Redis with 3 replicas.

### **Update**:

I have tested v1.5.0 and v1.5.1 locally and here are the results:

Test env:
- Single Redis container
- Env:
```
OS: Manjaro Linux x86_64 
Kernel: 6.1.55-1-MANJARO 
AMD Ryzen 9 6900HS with Radeon Graphics (16) @ 3.300GHz 
Host: ROG Zephyrus G15 GA503RW_GA503RW 1.0 
Memory: 16GB
```
- Load-test command:
```bash
$ livekit-cli load-test --url ws://localhost:7880 --api-key devkey --api-secret secret --room load-test --verbose --publishers 32 --subscribers 32  --identity-prefix test --duration 1m --video-resolution medium --video-codec h264 --num-per-second 10
```

```bash
$ livekit-cli load-test --url ws://localhost:7880 --api-key devkey --api-secret secret --room load-test2 --verbose --publishers 32 --subscribers 32  --identity-prefix test --duration 1m --video-resolution medium --video-codec h264 --num-per-second 10
```

V1.5.0 results:
```
info clients
$215
# Clients
connected_clients:113
cluster_connections:0
maxclients:10000
client_recent_max_input_buffer:20480
client_recent_max_output_buffer:0
blocked_clients:0
tracking_clients:0
clients_in_timeout_table:0
```

V1.5.1 results:
```
info clients
$216
# Clients
connected_clients:1137
cluster_connections:0
maxclients:10000
client_recent_max_input_buffer:20480
client_recent_max_output_buffer:0
blocked_clients:0
tracking_clients:0
clients_in_timeout_table:0
```

**Comment by davidzhao** [comment]
@itzloop can you share your config.yaml?

**Comment by itzloop** [comment]
> @itzloop can you share your config.yaml?

```yaml
port: 7880
prometheus_port: 8080
rtc:
  tcp_port: 7881
  port_range_start: 40000
  port_range_end: 60000
  use_external_ip: false
  node_ip: <node-public-ip>
  stun_servers:
    - stun.l.google.com:19302
    - stun1.l.google.com:19302
  turn_servers:
    - host: turn-domain
      port: 443
      protocol: udp
      username: username
      credential: password

redis:
  read_timeout: 10000
  dial_timeout: 5000
  write_timeout: 5000
  sentinel_addresses:
    - intance1:26379
    - intance2:26379
    - intance3:26379
    - intance4:26379
  sentinel_master_name: "my_master"
  sentinel_username: "user"
  sentinel_password: "supersecurepassword:)"
  password: "supersecurepassword:)"
turn:
    enabled: false
keys:
    apikey: secret

# Room Config
room:
  auto_create: true
  empty_timeout: 300 # 5 * 60 seconds
  enabled_codecs:
    - mime: "audio/opus"
    - mime: "audio/red"
    - mime: "video/VP8"
    - mime: "video/H264"
    - mime: "video/VP9"
    - mime: "video/AV1"

node_selector:
  kind: "any"
  sort_by: "bytespersec"
```

**Comment by paulwe** [comment]
thank you for your pr!

in 1.5.1 we added a psrpc version of the internal room and participant apis. the psrpc versions of these services register one pubsub topic per method rather than one per room/participant. the redis client creates one connection per topic because of restrictions redis imposes on clients while using pubsub. unfortunately this behavior ignores client pool size limits.

if you upgrade to the current master the redis message bus in psrpc has been updated to reuse one client connection for all subscriptions.

**Comment by itzloop** [comment]
@paulwe, thank you for the answer. I will test the master branch locally and confirm it. However, I have 2 questions:

1. Can you point me to the PR that got this issue fixed?
2. Does this mean I should close [#516](https://github.com/livekit/protocol/pull/516)? I still believe we should be able to configure pool size and timeout.

**Comment by paulwe** [comment]
1. it's fixed in #2239 but the relevant changes are in a separate repo - [diff](https://github.com/livekit/psrpc/compare/main..a8349d9fa62166cfa87d37412601d88bc4b2b00d)
2. it seems reasonable to have config for redis client pooling. these changes are orthogonal.

**Comment by itzloop** [comment]
This issue has been resolved with #2239. Closing this issue.
